conans/model/info.py

```diff
@@ -450,6 +454,10 @@ def package_id(self):
         result.append("")  # Append endline so file ends with LF
         text = '\n'.join(result)
         package_id = sha1(text.encode())
+        try:
+            self.options.validate()
+        except ConanException as e:
+            self.invalid = BINARY_INVALID, str(e)
```

Comment:
Recall me why you mark this as INVALID, I guess is for compatible packages, but put me an example.

Reply:
Yes, it is for compatible packages. Example: 
- ``cmake`` package built with all the settings, including compiler. Settings not removed in the ``package_id`` method.
- ``cmake`` package required with only ``os`` and ``arch`` setting. It will compute a pcakage_id which is invalid, won't be able to be installed or built
- But if you have a ``compatible_packages`` or ``compatibilitly.py`` that defines compatibles different compilers etc, it will work.